### PR TITLE
Fix/dependencies for php 8.1.x

### LINKED ISSUE #51: Removal of `laminas/laminas-zendframework-bridge`
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes
| BC Break    | no

#### Summary

_Originally posted by @boesing in https://github.com/laminas/technical-steering-committee/issues/91#issuecomment-891899044_

As per the latest TSC meeting from [2021-08-02](https://github.com/laminas/technical-steering-committee/blob/8f72cf6af75f22436d752ce7e7c0f92c23336666/meetings/agenda.md#remove-laminaslaminas-zendframework-bridge-dependency-from-our-packages), the removal is experimental and we think we can drop it in minor versions as of now. Since the skeleton is special, we should only remove the bridge from this project without having any issues.

As per @michalbundyra, this should have been a part of this project anyways, so removal should be okay.

### LINKED ISSUE #53: [RFC]: Removal of `Auryn` container
## RFC

| Q                   | A      |
| ------------------- | ------ |
| Proposed Version(s) | 3.11.0  |
| BC Break?           | No |

### Goal

The current implementation (https://packagist.org/packages/northwoods/container) seems to be unmaintained.
Last release was in 2018 (3 years ago) and it has an invalid implementation regarding invokables.

Current unit tests are failing due to a change in `mezzio` regarding PSR-17 response factories.

### Background

Invokables in a mezzio/laminas context are usually instantiated with the `InvokableFactory` which was introduced in `laminas-servicemanager` v3.0.0.

The `Auryn` container assumes that if a `invokables` configuration is provided, that this means that the `invokable` must be invokable.
So instead of instantiating the `invokable`, it checks if it `is_callable()`.

```php
private function makeInvokable(string $name, $factory): callable
{
    if (is_callable($factory)) {
        return $factory;
    }

    if ($this->isValidClass($factory)) {
        $factory = new $factory();
    }

    if (is_callable($factory) === false) {
        throw ContainerException::expectedInvokable($name);
    }

    return $factory;
}
```

### Considerations

We could provide a patch to that container but due to the lack of time, I would rather drop it. Even if we would provide a patch for this, there is no guarantee that it is still maintained.

### Proposal(s)

I would like to drop the `Auryn` container support with one of the next versions. Existing projects keep working, just freshly created projects are not able to select that container during installation.

### Appendix

None so far.

### PR DESCRIPTION
**Pull request by lukaslangen**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | unsure
| New Feature   | no
| RFC           | yes
| QA            | no

### Description

See #59

### CODE REVIEW

File: src/App/src/Handler/HomePageHandlerFactory.php [hunk 1]
```diff
@@ -20,6 +20,8 @@ public function __invoke(ContainerInterface $container): RequestHandlerInterface
             ? $container->get(TemplateRendererInterface::class)
             : null;
 
+        // @codingStandardsIgnoreStart
```
Comment (Ocramius) [inline src/App/src/Handler/HomePageHandlerFactory.php:23]: What was the reported issue here? :thinking:

**Review by Ocramius — APPROVED** [review]
LGTM.

@weierophinney @boesing can you check? I think having less integration work here is a good idea.

File: src/App/src/Handler/HomePageHandlerFactory.php (hunk 1 above)
Comment (lukaslangen) [reply src/App/src/Handler/HomePageHandlerFactory.php:23]:
```FILE: src/App/src/Handler/HomePageHandlerFactory.php
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
 23 | ERROR | [x] Class name referenced via call of function get_class().
    |       |     (SlevomatCodingStandard.Classes.ModernClassNameReference.ClassNameReferencedViaFunctionCall)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 81ms; Memory: 10MB
```

**Comment by brandonsavage** [comment]
I think removing Aura is a good call, since that project hasn't been updated in some time and doesn't appear that it will be. I'm migrating clients to Mezzio away from Aura, because it's not actively supported.

File: src/App/src/Handler/HomePageHandlerFactory.php (hunk 1 above)
Comment (arueckauer) [reply src/App/src/Handler/HomePageHandlerFactory.php:23]: `@codingStandardsIgnoreStart` is deprecated, `phpcs:disable` should be used. For a single line `phpcs:ignore` can be used, which causes a little less noise.
https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file

**Comment by Ocramius** [comment]
CS fixes are still running on 7.3: perhaps we just need to drop support for that?

**Review by weierophinney — CHANGES_REQUESTED** [review]
I agree with @Ocramius — bump the minimum supported PHP version to 7.4. From there, also do the following:

- Bump laminas-coding-standard to `^2.3` (and fix any issues raised).
- Bump PHPUnit to `^9.5.5`.
- Check other dependencies to determine which version first offers PHP 8.1 support, and bump to that when possible.
- Audit the cs-check and cs-fix scripts; is the `DealerDirect....Plugin::run` line really required for these anymore?

Before we release, I'd also like to see a separate PR that replaces phpstan with psalm, so that this package is consistent with our other packages.

**Comment by lukaslangen** [comment]
Okay. I've added the specifically requested new version constraints and checked all version constraints for `require` and adjusted them accordingly.

Also on the way I removed `laminas/laminas-zendframework-bridge` as per #51 

As for the cs-fix stuff: I don't have the time right now to go more deeply into this, but it seems, that this line is necessary to install *all* coding standards found in the vendor directory. It seems especially necessary for CI (at least after my understanding of the package's documentation, see [here](https://github.com/Dealerdirect/phpcodesniffer-composer-installer#calling-the-plugin-directly)

If it still should be removed, please just tell me and I'll do it as soon, as I have time for it.

**Comment by weierophinney** [comment]
> As for the cs-fix stuff: I don't have the time right now to go more deeply into this, but it seems, that this line is necessary to install _all_ coding standards found in the vendor directory. It seems especially necessary for CI (at least after my understanding of the package's documentation, see [here](https://github.com/Dealerdirect/phpcodesniffer-composer-installer#calling-the-plugin-directly)

What's odd is we literally don't have that in any other package we ship, and CI works just fine. :)

I have time today, so I'll see if I can push that change and fixes for the failing unit tests on 7.4 with lowest deps.

**Review by weierophinney — APPROVED** [review]

**Pull request merged**